lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -260,37 +260,40 @@ def compile_subconstants(name, constant)
           "\n" + output.join("\n\n")
         end
 
-        sig { params(value: T.untyped).returns(String) }
-        def compile_type_variable_args(value)
-          parts = []
-          parts << ":#{value.variance}" unless value.variance == :invariant
-          parts << "fixed: #{value.fixed}" if value.fixed
-          parts << "lower: #{value.lower}" if value.lower
-          parts << "upper: #{value.upper}" if value.upper
-
-          parts.join(", ")
-        end
-
         sig { params(name: String, constant: Module).returns(T.nilable(String)) }
         def compile_type_variables(name, constant)
           with_indentation do
+            # Try to find the type variables defined on this constant, bail if we can't
             type_variables = constant.respond_to?(:__type_variables) && T.unsafe(constant).__type_variables
-            return unless type_variables
-
-            type_constants_with_index = constants_of(constant).sort.uniq.map do |constant_name|
-              symbol = "#{name}::#{constant_name}"
-              subconstant = resolve_constant(symbol)
-              index = type_variables.index(subconstant)
-              [constant_name, subconstant, index] if index
-            end.compact
-
-            output = type_constants_with_index.sort_by(&:last).map do |constant_name, value|
-              if T::Types::TypeMember === value
-                indented("#{constant_name} = type_member(#{compile_type_variable_args(value)})")
-              elsif T::Types::TypeTemplate === value
-                indented("#{constant_name} = type_template(#{compile_type_variable_args(value)})")
+            return unless type_variables && Array === type_variables
+
+            # For each subconstant defined under this constant
+            # we resolve it to a subconstant object and look up its index the
+            # value of it in the type variable list that we have collected.
```

Comment:
Seems like something is missing here: `look up its index the value of it`

Reply:
These comments were refactored heavily in later commits, so this was probably fixed.